js-core/phonebook/src/main.js

```diff
@@ -0,0 +1,99 @@
+/*
+Приложение телефонный справочник
+
+Создайте функцию конструктор.
+У данной функции должны быть методы:
+
+1 Преобразование телефонного номера из формата 0993378130 в (099) 33-78-130
+2 Проверка, что телефонный номер содержит только числа
+3 Добавление пользователей в справочник
+4 Удаление пользователя по имени, фамилии
+5 Поиск пользователей по имени - отображает всех пользователей с одинаковым именем
+6 Изменение имени, фамилии, телефонного номера у выбраного пользователя ( здесь должно быть реализовано через this )
+7 Сортировка пользователей по номеру телефона, фамилии, имени и тд, по любому из свойств пользователя
+8 Фильтр по указанному свойству
+
+*/
+
+
+class PhoneApp {
+    constructor() {
+        this.dataBase = [
+            {firstName:'Оля', secondName: 'Кузьмина', phone:'0661234567'},
+            {firstName:'Катя', secondName: 'Иванова', phone:'0991234567'},
+            {firstName:'Оля', secondName: 'Петрова', phone:'0631234567'},
+            {firstName:'Марина', secondName: 'Сидорова',phone:'0951234567'}
+        ]
+    }
+    /* 1. Преобразование телефонного номера из формата 0993378130 в (099) 33-78-130 */
+    conversionPhoneFormat (phoneNumber) {
+        let str = '';
+        str += phoneNumber;
+        return "(" + str.substring(0,3) + ")" + " " + str.substring(3,5) + "-" +
+            str.substring(5,7) + "-" + str.substring(7,10);
+    }
+    /* 2. Проверка, что телефонный номер содержит только числа */
+    chekNumber (phoneNumber) {
+        return +phoneNumber ? true : false;
+    };
+    /* 3. Добавление пользователей в справочник */
+    addUser (newUser) {
+        this.dataBase.push(newUser);
+
+    };
+    /* 4. Удаление пользователя по имени, фамилии */
+    deleteUser (name) {
+        for (let i = 0; i < this.dataBase.length; i++) {
+            let elem = this.dataBase[i];
+            if (elem.firstName === name || elem.secondName === name) {
+                this.dataBase.splice(i, 1);
```

Comment:
splice not really safe. I would find a better solution, you can use Array.prototyp.filter for example